crates/sui-core/src/authority/epoch_start_configuration.rs

```diff
@@ -74,6 +107,28 @@ impl EpochStartConfigurationV1 {
     }
 }
 
+impl EpochStartConfigurationV2 {
+    pub fn new(
+        system_state: EpochStartSystemState,
+        epoch_digest: CheckpointDigest,
+        flags: Vec<EpochFlag>,
+    ) -> Self {
+        Self {
+            system_state,
+            epoch_digest,
+            flags,
+        }
+    }
+
+    pub fn new_for_testing() -> Self {
+        Self::new(
+            EpochStartSystemState::new_for_testing(),
+            CheckpointDigest::default(),
+            vec![],
```

Comment:
should this use default_for_new_epoch?

Reply:
Yes! Changed it here - https://github.com/MystenLabs/sui/pull/10772/files#diff-1b9896aa231b63aa93d694e6cd03aee642d0878f6c5b368aed4705dc6a95e7cbR60